### PR TITLE
[DOC-10097] 24.1 features that require upgrade finalization

### PR DESCRIPTION
**Pull request by mdlinville**
[DOC-10097] 24.1 features that require upgrade finalization

Src: https://github.com/cockroachdb/cockroach/blob/release-24.1.0-rc/pkg/clusterversion/cockroach_versions.go#L260

### Preview
[src/current/cockroachcloud/upgrade-to-v24.1.md](https://deploy-preview-18532--cockroachdb-docs.netlify.app/docs/cockroachcloud/upgrade-to-v24.1.html)
[src/current/v24.1/upgrade-cockroach-version.md](https://deploy-preview-18532--cockroachdb-docs.netlify.app/docs/v24.1/upgrade-cockroach-version.html)

### CODE REVIEW

File: src/current/cockroachcloud/upgrade-to-v24.1.md [hunk 1]
```diff
@@ -134,7 +134,9 @@ Use the [DB Console]({% link cockroachcloud/tools-page.md %}) or your own toolin
 
 ### Expect temporary limitations
 
-Most {{ page.page_version }} features can be used right away, but some will be enabled only after the upgrade has been finalized. Attempting to use these features before finalization will result in errors. {% comment %}TODO: List of temporary limitations for GA{% endcomment %}
+Most {{ page.page_version }} features can be used right away, but some will be enabled only after the upgrade has been finalized. Attempting to use these features before finalization will result in errors.
+
+- Splits no longer hold latches for time proportional to the range size while computing MVCC statistics. Instead, MVCC statistics are pre-computed before the critical section of the split. As a side effect, the resulting statistics are no longer 100% accurate because they may correctly distribute writes concurrent with the split. To mitigate against this potential inaccuracy, and to prevent the statistics from drifting after successive splits, the existing stored statistics are re-computed and corrected if needed during the non-critical section of the split. [#119894](https://github.com/cockroachdb/cockroach/pull/119894)
```
Comment (rmloveland) [inline src/current/cockroachcloud/upgrade-to-v24.1.md:139]: 'splits' can link to https://www.cockroachlabs.com/docs/v24.1/architecture/distribution-layer.html#range-splits

'latches' can link to https://www.cockroachlabs.com/docs/v24.1/architecture/transaction-layer#latch-manager

'mvcc' can link to https://www.cockroachlabs.com/docs/v23.2/architecture/storage-layer.html#mvcc 

this is duplicated with the other page and should be an include

File: src/current/v24.1/upgrade-cockroach-version.md [hunk 2]
```diff
@@ -135,18 +135,9 @@ By default, after all nodes are running the new version, the upgrade process wil
...
-- A fix to a bug that could allow a user to execute a user-defined function without the `EXECUTE` privilege on the function. If a user does not have the privilege, the user-defined function does not run and an error is logged. [#114203][#114203]
-
-For more details about a given feature, refer to the [CockroachDB v23.2.0 release notes](https://www.cockroachlabs.com/docs/releases/v23.2#v23-2-0).
+- Splits no longer hold latches for time proportional to the range size while computing MVCC statistics. Instead, MVCC statistics are pre-computed before the critical section of the split. As a side effect, the resulting statistics are no longer 100% accurate because they may correctly distribute writes concurrent with the split. To mitigate against this potential inaccuracy, and to prevent the statistics from drifting after successive splits, the existing stored statistics are re-computed and corrected if needed during the non-critical section of the split. [#119894](https://github.com/cockroachdb/cockroach/pull/119894)
```
Comment (rmloveland) [inline src/current/v24.1/upgrade-cockroach-version.md:138]: see my previous comment re: suggested links, and that this should be an include

**Review by rmloveland — APPROVED** [review]
LGTM with non-blocking suggestions

Suggest getting technical review on this as well.  I assume maybe you've had convos elsewhere with engineering team folks re: these are the features that require upgrade finalization?  I realize you shared a link to the code but like, idk how to map from the code in that file to what should be in this doc so I'm assuming you do and/or have talked to people who do.  I also looked at [old docs PRs for "upgrade finalization"](https://github.com/cockroachdb/docs/pulls?q=is%3Apr+is%3Aclosed+upgrade+finalization+) and discovered that a lot of those PRs didn't have eng reviewers so maybe I'm just wrong and it's not necessary?

**Comment by mdlinville** [comment]
> LGTM with non-blocking suggestions
> 
> Suggest getting technical review on this as well. I assume maybe you've had convos elsewhere with engineering team folks re: these are the features that require upgrade finalization? I realize you shared a link to the code but like, idk how to map from the code in that file to what should be in this doc so I'm assuming you do and/or have talked to people who do.
> 
> I also looked at [old docs PRs for "upgrade finalization"](https://github.com/cockroachdb/docs/pulls?q=is%3Apr+is%3Aclosed+upgrade+finalization+) and discovered that a lot of those PRs didn't have eng reviewers so maybe I'm just wrong and it's not necessary? Defer to your expertise in this area ofc but that's just what came to my mind

Thanks. In brief, each line in that section of that file was added by an engineer because their feature requires a migration (and thus, finalization). So we go to the correct area of the file and then use Github's handy Git Blame view to see which commit added each line. From the commit I can get to the PR, where I can read about it and find the engineer who authored it or reviewed it and ask them some questions. That's why I was asking you about protected timestamps earlier. :)

**Comment by rmloveland** [comment]
>     LGTM with non-blocking suggestions
>
>     Suggest getting technical review on this as well. I assume maybe
>     you've had convos elsewhere with engineering team folks re: these
>     are the features that require upgrade finalization? I realize you
>     shared a link to the code but like, idk how to map from the code
>     in that file to what should be in this doc so I'm assuming you do
>     and/or have talked to people who do.
>
>     I also looked at old docs PRs for "upgrade finalization" and
>     discovered that a lot of those PRs didn't have eng reviewers so
>     maybe I'm just wrong and it's not necessary? Defer to your
>     expertise in this area ofc but that's just what came to my mind

> Thanks. In brief, each line in that section of that file was added by
> an engineer because their feature requires a migration (and thus,
> finalization). So we go to the correct area of the file and then use
> Github's handy Git Blame view to see which commit added each line.
> From the commit I can get to the PR, where I can read about it and
> find the engineer who authored it or reviewed it and ask them some
> questions. That's why I was asking you about protected timestamps
> earlier. :)

Makes sense, thanks for explaining this process to me!  (I haven't done
it before myself)

**Pull request merged**